src/surge-xt/SurgeSynthProcessor.cpp

```diff
@@ -452,6 +479,25 @@ void SurgeSynthProcessor::processBlockMidiFromGUI()
     }
 }
 
+void SurgeSynthProcessor::processBlockOSC()
+{
+    while (true)
```

Comment:
So, this is why the RingBuffer class has a popall. Suggest a slight logic change.

```
auto messages = oscRingBuf.popall();
for (const auto& m : messages) {
  float pval = m.val;
  // etc...
}
```